### PR TITLE
Migrate Gateway instances screen to Angular

### PR DESCRIPTION
**Pull request by gaetanmaisse**
## Issue

https://gravitee.atlassian.net/browse/APIM-3366

## Description

Migrate Gateway instances screen to Angular. 
Sorry for the commit history, I'm a bit rusty.

UX changes:
 - The search box has been removed as it was not working 
 - A Pagination has been put in place as it was already implemented on the backend side
 - All instances are shown, whatever their status

**Before**
Capture d’écran 2023-11-17 à 12 06 01

**After**
Capture d’écran 2023-11-17 à 12 06 43

With tenant and tags:
Capture d’écran 2023-11-17 à 10 19 10

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-gplbmgvyrg.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/instances/_instances.scss [hunk 1]
```diff
@@ -1,59 +0,0 @@
...
-  a:link,
-  a:visited,
-  a:hover,
-  a:active {
-    text-decoration: none;
-    color: black;
-  }
-
-  md-card-header {
-    padding: 0px 0px 20px 0px;
```
Comment (jansobczynski) [inline gravitee-apim-console-webui/src/management/instances/_instances.scss:26]: Might be simplify to `padding: 0 0 20px;`

File: gravitee-apim-console-webui/src/management/instances/_instances.scss [hunk 2]
```diff
@@ -1,59 +0,0 @@
...
-  }
-  md-card-avatar {
-    width: 35px;
-    height: 20px;
-  }
-  md-card-content {
-    padding: 0px;
-  }
-  md-card-footer {
-    padding: 0px 5px 5px 5px;
```
Comment (jansobczynski) [inline gravitee-apim-console-webui/src/management/instances/_instances.scss:36]: The same here `padding: 0 5px 5px;`

File: gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.component.html [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          size="20"
+          svgIcon="gio:stop-circle"
+        ></mat-icon>
+        <mat-icon
+          *ngIf="element.state !== 'STOPPED' && element.state !== 'STARTED'"
+          matTooltip="Unknown"
+          class="state__instance-unknown"
+          size="20"
+          svgIcon="gio:question-mark-circle"
+        ></mat-icon>
```
Comment (jansobczynski) [inline gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.component.html:66]: Do we have only 3 element.state properties? Could we just use here `element.state === 'UNKNOWN'`?

File: gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.component.scss [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  width: 30%;
+}
+
+.mat-column-version {
+  width: 10%;
+}
+
+.mat-column-state {
+  width: 1%;
+  padding: 0 16px 0 0;
```
Comment (jansobczynski) [inline gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.component.scss:22]: We can simplify to `padding: 0 16px 0;`

File: gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.module.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { MatTableModule } from '@angular/material/table';
+import { MatIconModule } from '@angular/material/icon';
+import { MatTooltipModule } from '@angular/material/tooltip';
+import { UIRouterModule } from '@uirouter/angular';
+
+import { InstanceListComponent } from './/instance-list.component';
```
Comment (jansobczynski) [inline gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.module.ts]: `'.//instance-list.component';` double slash typo?

File: gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.module.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+import { GioTableWrapperModule } from '../../../shared/components/gio-table-wrapper/gio-table-wrapper.module';
+
+@NgModule({
+  declarations: [InstanceListComponent],
+  exports: [InstanceListComponent],
+  imports: [CommonModule, MatTableModule, GioTableWrapperModule, MatIconModule, MatTooltipModule, UIRouterModule],
+})
```
Comment (jansobczynski) [inline gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.module.ts:32]: We are trying to keep it clean, so by the order, MatModules all next to each other :)

File: gravitee-apim-console-webui/src/management/instances/instances.route.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+import { InstanceListComponent } from './instance-list/instance-list.component';
+import { InstanceDetailsComponent } from './instance-details/instance-details.component';
+import { InstanceDetailsEnvironmentComponent } from './instance-details/instance-details-environment/instance-details-environment.component';
+import { InstanceDetailsMonitoringComponent } from './instance-details/instance-details-monitoring/instance-details-monitoring.component';
+
+export const states: Ng2StateDeclaration[] = [
+  {
+    name: `management.instances-list`,
+    url: '/instances',
+    component: InstanceListComponent,
```
Comment (jansobczynski) [inline gravitee-apim-console-webui/src/management/instances/instances.route.ts:27]: I'm wondering if we shouldn't call it `component: InstancesListComponent,`?

Comment (gaetanmaisse) [reply gravitee-apim-console-webui/src/management/instances/instances.route.ts:27]: I've matched what was done in API and Application so no strong opinions on this :)

File: gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.component.html [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+      </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="this.displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: this.displayedColumns"></tr>
+
+    <!-- Row shown when there is no data -->
+    <tr class="mat-row" *matNoDataRow>
+      <td class="mat-cell" [attr.colspan]="this.displayedColumns.length">
+        {{ 'There is no Gateway (yet).' }}
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.component.html]:
```suggestion
        {{ 'There are no Gateway instances (yet).' }}
```

**Review by ytvnr — APPROVED** [review]
just a grammar issue

**Review by jansobczynski — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.component.html (hunk 8 above)
Comment (gaetanmaisse) [reply gravitee-apim-console-webui/src/management/instances/instance-list/instance-list.component.html]: Fixed

**Review by ThibaudAV — APPROVED** [review]

**Pull request merged**